third_party/machinarium/sources/context.c

```diff
@@ -64,41 +72,71 @@ void mm_context_create(mm_context_t *context, mm_contextstack_t *stack,
 	mm_context_swap(&context_runner, context);
 }
 
-#if !defined(__amd64) && !defined(__i386)
+#if !defined(__aarch64__) && !defined(__amd64) && !defined(__i386)
 #error unsupported architecture
 #endif
 
-asm("\t.text\n"
-    "\t.globl mm_context_swap\n"
```

Comment:
First two asm lines are still common. Did you copied them intentionally so that for each arch we have valid ASM chunk?

Reply:
yes